### ISSUE TITLE
runtime error: index out of range

### ISSUE THREAD

**Issue by francesco-plt** [paij0se/cligpt#2]
Followed the instructions and installed by running `go install github.com/paij0se/cligpt@latest`. I cannot launch the cli tool because it always results in the following error: `panic: runtime error: index out of range [1] with length xxx`. I'm running it on an M1 Macbook Pro with macOS 13.1.

**Comment by paij0se** [comment]
Hey francesco, you inserted the OpenAI token? If so, it could be:

- You don't have credits.
- The token is incorrect.

Please verify if the token is correct.

![image](https://user-images.githubusercontent.com/69026987/209689494-c151b648-f2e6-4f19-92e0-2b155844acd2.png)

**Comment by paij0se** [comment]
Check `$HOME/.config/cligpt/cligpt.yml`

**Comment by francesco-plt** [comment]
I tried reinstalling it and writing manually the API key to `$HOME/.config/cligpt/cligpt.yml`, but the result is still the same:
<img width="888" alt="Screenshot 2022-12-27 alle 22 49 48" src="https://user-images.githubusercontent.com/26849744/209725469-203f2132-5339-4827-9af5-c91bb70e5ccc.png">

**Comment by paij0se** [comment]
Ok man, try the new version `v1.0.2`

this version will show you if the token is valid or not.

**Comment by sasha-wing** [comment]
I get 
```
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
main.main()
	/Users/sasha/go/pkg/mod/github.com/paij0se/cligpt@v1.0.3/main.go:86 +0x851
```
my `/.config/cligpt/cligpt.yml`
```
auth: token
model: text-davinci-003
max_tokens: 256
```

**Comment by paij0se** [comment]
@sasha-wing Make sure you have credits.

**Comment by francesco-plt** [comment]
> Ok man, try the new version `v1.0.3`
> 
> this version will show you if the token is valid or not.

Tried updating, but no changes so far

**Comment by n14s** [comment]
Same for me...

**Comment by RagingTiger** [comment]
Same here

**Comment by RagingTiger** [comment]
> Same here:
> 
> ```
> Please write your token: API_KEY
> panic: runtime error: index out of range [1] with length 1
> 
> goroutine 1 [running]:
> main.main()
>         /go/pkg/mod/github.com/paij0se/cligpt@v1.0.3/main.go:54 +0x85e
> ```

Never mind figured it out.

**Comment by francesco-plt** [comment]
> > Same here:
> > ```
> > Please write your token: API_KEY
> > panic: runtime error: index out of range [1] with length 1
> > 
> > goroutine 1 [running]:
> > main.main()
> >         /go/pkg/mod/github.com/paij0se/cligpt@v1.0.3/main.go:54 +0x85e
> > ```
> 
> Never mind figured it out.

Mind sharing your solution?

**Comment by RagingTiger** [comment]
> > > Same here:
> > > ```
> > > Please write your token: API_KEY
> > > panic: runtime error: index out of range [1] with length 1
> > > 
> > > goroutine 1 [running]:
> > > main.main()
> > >         /go/pkg/mod/github.com/paij0se/cligpt@v1.0.3/main.go:54 +0x85e
> > > ```
> > 
> > 
> > Never mind figured it out.
> 
> Mind sharing your solution?

First make sure you have an `API_KEY` as mentioned here: https://beta.openai.com/docs/quickstart/add-your-api-key.
Then make sure you call `cligpt` WITH your *prompt*:
```
cligpt 'How does ChatGPT API work?'
```
Basically, the above error isn't about the API_KEY, it's about NOT receiving the correct CLI args (i.e. the *prompt string*). 
In other words, simply calling `cligpt` without arguments (assuming you've already generated a config file at `/$HOME/.config/cligpt/cligpt.yml`) will fail with the above error:
```
# this will fail even if config is setup with correct API_KEY
$ chatgpt

# this will succeed assuming you have setup your config file correctly with the right API_KEY
$ chatgpt 'Please write a haiku about war'
```
Seems the author didn't handle the *case* where users **don't pass** a *prompt string*. Should be an easy fix, hehe. Does that make sense?

Let me know.

**Comment by paij0se** [comment]
Thank you @RagingTiger! I'm going to make a solution for the problem hehehe

**Comment by paij0se** [comment]
fixed https://github.com/paij0se/cligpt/releases/tag/v1.0.4

**Comment by falcucci** [comment]
I'll leave a comment just to let you know that this error can still happen if you have double quotes or special characters in your question e.g a `git diff`.

you can reproduce this problem at #6

**Comment by paij0se** [comment]
Fixed in the new [v1.0.5](https://github.com/paij0se/cligpt/releases/tag/v1.0.5) version.
